client/homescreen/activity-panel/orders/index.js

```diff
@@ -304,12 +304,8 @@ export default withSelect( ( select, props ) => {
 		REPORTS_STORE_NAME
 	);
 
-	if ( countUnreadOrders === null ) {
-		return { isRequesting: true };
-	}
-
-	if ( countUnreadOrders === 0 ) {
-		return { isRequesting: false };
+	if ( ! countUnreadOrders ) {
```

Comment:
so to clarify, the conditions are:

1. countUnreadOrders === null,  isRequesting: false
2. countUnreadOrders === 0, isRequesting: false
3. countUnreadOrders !== null && countUnreadOrders > 0, isRequesting: true

Is that right? I would have thought that `null` would mean that countUnreadOrders has not resolved yet here?

Reply:
When `countUnreadOrders` is resolving, it's value will be `null`. So when `! countUnreadOrders` is `true`, we ask if `countUnreadOrders` is not 0 in order to know that it's resolving.

Reply:
So the possible paths are:
```
if ( ! countUnreadOrders ) {
     return { isRequesting: countUnreadOrders !== 0 };
}
```
1. countUnreadOrders === null, isRequesting: `true`
2. countUnreadOrders === 0, isRequesting: `false`
3. countUnreadOrders !== null && countUnreadOrders > 0, it shouldn't get into the `if`

Reply:
Ah you're right, I misread the logic. 🤦 looking good